### PR TITLE
disable combine operations between different types of path objects

### PR DESCRIPTION
**Pull request by pineapplespatula**
fix for issue #3163

### CODE REVIEW

**Comment by volfied** [comment]
While this is correct, I had to go in `layerstructure.js` to see how `selectedLayersCanHaveVectorMask` is important to disabling combine operations. I think you should instead add a layerstructure property to make sure all selected layers are shape layers. It's better to check for 1 type than n-1 types.

**Comment by pineapplespatula** [comment]
I added a selectedLayersContainOnlyShapeLayers function which seems to be a little simpler

File: src/js/models/layerstructure.js [hunk 1]
```diff
@@ -586,6 +586,17 @@ define(function (require, exports, module) {
...
                 !curLayer.locked;
+        },
+
+        /**
+         * determine if there are only Shape layers in the current selection
+         *
+         * @return {boolean} if the layer selection contains only vector shapes
+         */
+        "selectedLayersContainOnlyShapeLayers": function () {
+            return this.selected.all(function (layer) {
```
Comment (volfied) [inline src/js/models/layerstructure.js]: Can you use `.every()` here, even if .all works, it's not documented anywhere, and is not conventional. We use `every` everywhere else.

**Comment by volfied** [comment]
Back to you @pineapplespatula

**Comment by pineapplespatula** [comment]
@volfied  :repeat:

**Pull request merged**